src/deepsparse/benchmark/benchmark_model.py

```diff
@@ -303,94 +307,131 @@ def parse_num_streams(num_streams, num_cores, scenario):
             return default_num_streams
 
 
-def main():
-
-    args = parse_args()
-
-    if args.quiet:
+def benchmark_model(
+    model_path: str,
+    batch_size: str = 1,
+    input_shapes: str = "",
+    num_cores: int = None,
+    scenario: str = "sync",
+    time: int = 10,
+    warmup_time: int = 2,
+    num_streams: int = None,
+    thread_pinning: str = "core",
+    engine: str = DEEPSPARSE_ENGINE,
+    quiet: bool = False,
+    export_path: str = None,
+) -> Dict:
+    if quiet:
         set_logging_level(logging.WARN)
 
-    decide_thread_pinning(args.thread_pinning)
+    decide_thread_pinning(thread_pinning)
 
-    scenario = parse_scenario(args.scenario.lower())
+    scenario = parse_scenario(scenario.lower())
     scheduler = parse_scheduler(scenario)
-    input_shapes = parse_input_shapes(args.input_shapes)
+    input_shapes = parse_input_shapes(input_shapes)
 
-    orig_model_path = args.model_path
-    args.model_path = model_to_path(args.model_path)
+    orig_model_path = model_path
+    model_path = model_to_path(model_path)
 
     # Compile the ONNX into a runnable model
-    if args.engine == DEEPSPARSE_ENGINE:
+    if engine == DEEPSPARSE_ENGINE:
         model = compile_model(
-            model=args.model_path,
-            batch_size=args.batch_size,
-            num_cores=args.num_cores,
-            num_streams=args.num_streams,
+            model=model_path,
+            batch_size=batch_size,
+            num_cores=num_cores,
+            num_streams=num_streams,
             scheduler=scheduler,
             input_shapes=input_shapes,
         )
         num_streams = parse_num_streams(model.num_streams, model.num_cores, scenario)
-    elif args.engine == ORT_ENGINE:
+    elif engine == ORT_ENGINE:
```

Comment:
Could use an else statement with a `raise ValuError` to prevent an undefined `model` variable further down.

Reply:
good call thanks